pkg/builder/local_build_executor_test.go

```diff
@@ -646,12 +667,44 @@ func TestLocalBuildExecutorWithWorkingDirectorySuccess(t *testing.T) {
 	contentAddressableStorage.EXPECT().PutFile(ctx, buildDirectory, "stderr", gomock.Any()).Return(
 		digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000006", 678),
 		nil)
+
+	inputRootDirectory.EXPECT().EnterDirectory("outputParent").Return(outputParentDirectory0, nil)
+	outputParentDirectory0.EXPECT().EnterDirectory("output").Return(outputDirectory0, nil)
+	outputParentDirectory0.EXPECT().Close()
+	outputDirectory0.EXPECT().Lstat("hello.pic.d").Return(
+		filesystem.NewFileInfo("hello.pic.d", filesystem.FileTypeRegularFile),
+		nil)
+	outputDirectory0.EXPECT().Close()
 	contentAddressableStorage.EXPECT().PutFile(ctx, outputDirectory0, "hello.pic.d", gomock.Any()).Return(
 		digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000007", 789),
 		nil)
+
+	inputRootDirectory.EXPECT().EnterDirectory("outputParent").Return(outputParentDirectory1, nil)
+	outputParentDirectory1.EXPECT().EnterDirectory("output").Return(outputDirectory1, nil)
+	outputParentDirectory1.EXPECT().Close()
+	outputDirectory1.EXPECT().EnterDirectory("foo").Return(fooDirectory, nil)
+	outputDirectory1.EXPECT().Close()
+	fooDirectory.EXPECT().EnterDirectory("objects").Return(objectsDirectory, nil)
+	fooDirectory.EXPECT().Close()
+	objectsDirectory.EXPECT().Lstat("hello.pic.o").Return(
+		filesystem.NewFileInfo("hello.pic.o", filesystem.FileTypeExecutableFile),
+		nil)
+	objectsDirectory.EXPECT().Close()
 	contentAddressableStorage.EXPECT().PutFile(ctx, objectsDirectory, "hello.pic.o", gomock.Any()).Return(
 		digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000008", 890),
 		nil)
+
+	inputRootDirectory.EXPECT().EnterDirectory("outputParent").Return(outputParentDirectory0, nil)
+	outputParentDirectory0.EXPECT().Lstat("..").Return(
```

Comment:
You can't use the filesystem.Directory API to traverse upwards in the file system hierarchy.

Reply:
Tracked down the missing path.Join before a path.Dir. Also cleaned up the error handling in this scenario and added a test for it specifically, because this isn't obvious and eludes the rest of the tests.